lib/adapter/codecept.js

```diff
@@ -58,14 +63,45 @@ function CodeceptReporter(config) {
     // clear tmp dir
     fileSystem.clearDir(TESTOMAT_TMP_STORAGE_DIR);
 
-    recorder.add('Creating new run', () => client.createRun());
+    // recorder.add('Creating new run', () => );
+    client.createRun();
     videos = [];
     traces = [];
 
     if (!global.testomatioDataStore) global.testomatioDataStore = {};
   });
 
-  event.dispatcher.on(event.test.before, () => {
+  let hookSteps = [];
+  let suiteHookRunning = false;
+
+  event.dispatcher.on(event.suite.before, suite => {
+    suiteHookRunning = true;
+    hookSteps = [];
+    global.testomatioDataStore.steps = [];
+
+    logger.setContext(suite.fullTitle());
+  });
+
+  event.dispatcher.on(event.suite.after, () => {
+    logger.setContext(null);
+  });
+
+  event.dispatcher.on(event.hook.started, suite => {
+    // global.testomatioDataStore.steps = [];
+  });
+
+  event.dispatcher.on(event.hook.passed, suite => {
+    if (suiteHookRunning) hookSteps.push(...global.testomatioDataStore.steps);
```

Comment:
Here we need to stop listening for BeforeSuite messages:

```js
if (suiteHookRunning) logger.setContext(null);
```
As everything beyond that point will go to a test log